staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -474,11 +475,18 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 
 	// Determine watch timeout('0' means deadline is not set, ignore checking)
 	deadline, _ := ctx.Deadline()
+
+	userAgent := "unknown"
+	if requestInfo, ok := request.RequestInfoFrom(ctx); ok {
+		userAgent = requestInfo.UserAgent
```

Comment:
why user agent and not username?

Reply:
or audit id

Reply:
I wanted to make it possible to match the watch with httplog.go line like: 

```
I0111 11:36:19.019239      12 httplog.go:89] "HTTP" verb="GET" URI="/api/v1/namespaces/test-lir0yp-1/secrets?allowWatchBookmarks=true&fieldSelector=metadata.name%3Ddefault-token-xw8pq&resourceVersion=3180522&timeout=7m38s&timeoutSeconds=458&watch=true" latency="7m38.000749718s" userAgent="kubelet/v1.21.0 (linux/amd64) kubernetes/a833f0b" srcIP="10.40.5.204:49050" resp=0
```

We have user-agent, not username there.

I dropped user-agent from this, as I think that key (~url), label and field selector will be sufficient in most of the cases.